search-provider/eks-article-metadata-query-provider.c

```diff
@@ -0,0 +1,431 @@
+                                     state->provider->application_id,
+                                     result,
+                                     (GSList **) &models,
+                                     (GSList **) &shards,
+                                     &error))
+    {
+      g_dbus_method_invocation_take_error (state->invocation,
+                                           g_steal_pointer (&error));
+      return;
+    }
+
+  for (GSList *l = (GSList *) models; l; l = l->next)
+    {
+      EkncContentObjectModel *model = l->data;
+
+      g_variant_builder_open (&builder, G_VARIANT_TYPE_VARDICT);
+
+      maybe_add_key_value_pair_from_model_to_variant (model,
+                                                      &builder,
+                                                      "title",
+                                                      G_VARIANT_TYPE_STRING);
+      maybe_add_key_value_pair_from_model_to_variant (model,
+                                                      &builder,
+                                                      "content_type",
+                                                      G_VARIANT_TYPE_STRING);
+      maybe_add_key_value_pair_from_model_to_variant (model,
+                                                      &builder,
+                                                      "thumbnail_uri",
+                                                      G_VARIANT_TYPE_STRING);
+      maybe_add_key_value_pair_from_model_to_variant (model,
+                                                      &builder,
+                                                      "ekn_id",
+                                                      G_VARIANT_TYPE_STRING);
+      maybe_add_key_value_pair_from_model_to_variant (model,
+                                                      &builder,
+                                                      "child_tags",
+                                                      G_VARIANT_TYPE_STRING_ARRAY);
+      maybe_add_key_value_pair_from_model_to_variant (model,
+                                                      &builder,
+                                                      "tags",
+                                                      G_VARIANT_TYPE_STRING_ARRAY);
+
+      g_variant_builder_close (&builder);
+    }
+
+  shards_strv = strv_from_shard_list ((GSList *) shards);
+
+  eks_article_metadata_complete_query (state->provider->skeleton,
+                                       state->invocation,
+                                       (const gchar * const *) shards_strv,
+                                       g_variant_builder_end (&builder));
+}
+
+static void
+append_construction_prop_from_string (const gchar *key,
+                                      const gchar *str,
+                                      GArray      *values_array,
+                                      GPtrArray   *props_array)
+{
+  guint index = values_array->len;
+  GValue *value = NULL;
```

Comment:
In general, try to take advantage of C99-style declarations; no need to declare variables only at the top, you can move this down to where it gets its actual value.

Reply:
Okay. I didn't realize that C99-style was permissible here.

Reply:
In the meantime, to avoid churn, I'd prefer to just keep this file consistent the way it is, if that's okay with you.